### PR TITLE
Feature/changelog for subquery splicing

### PR DESCRIPTION
**Pull request by markuspf**
This adds a CHANGELOG entry for the recent work on AQL execution optimisation by splicing subqueries

### CODE REVIEW

File: CHANGELOG [hunk 1]
```diff
@@ -1,6 +1,29 @@
...
+
+	This optimisation splices the execution of subqueries inside the
+	execution of their surrounding query. For subqueries with a fairly
+	small number of returned rows this saves a significant amount of
+	overhead initializing cursors.
+
+	This optimisation is performed by the optimizer rule
+	`splice-subqueries`, and currently only works on suitable
+	subqueries; A subquery becomes unsuitable if it contains a LIMIT,
+	REMOTE, GATHER or a COLLECT node where the operation is not COUNT.
```
Comment (goedderz) [inline CHANGELOG]: > or a COLLECT node where the operation is not COUNT

It's the other way round, `COLLECT` with `COUNT` makes the query unsuitable, the other variants do not.

File: CHANGELOG [hunk 2]
```diff
@@ -1,6 +1,29 @@
...
+	unsuitable subquery.
+
+	The implementation of this feature required reworking the dataflow
+  of query execution.
+
+* Rewrote all modification executors
+
+	as a consequence of the splicing subquery optimisation we rewrote
+	all modification executors. No functional change should have
+	happened in the process.
```
Comment (goedderz) [inline CHANGELOG]: As this is not a visible change, I think it does not belong in the CHANGELOG.

File: CHANGELOG [hunk 3]
```diff
@@ -1,6 +1,29 @@
 devel
 -----
 
+* Added a new subquery optimization called `splice-subqueries`.
+
+	This optimisation splices the execution of subqueries inside the
```
Comment (Simran-B) [inline CHANGELOG]:
```suggestion
	This optimization splices the execution of subqueries inside the
```

File: CHANGELOG [hunk 4]
```diff
@@ -1,6 +1,29 @@
...
+	execution of their surrounding query. For subqueries with a fairly
+	small number of returned rows this saves a significant amount of
```
Comment (Simran-B) [inline CHANGELOG]: rows -> documents?

File: CHANGELOG [hunk 5]
```diff
@@ -1,6 +1,29 @@
...
+	overhead initializing cursors.
+
+	This optimisation is performed by the optimizer rule
```
Comment (Simran-B) [inline CHANGELOG]:
```suggestion
	This optimization is performed by the optimizer rule
```

File: CHANGELOG [hunk 6]
```diff
@@ -1,6 +1,29 @@
...
+	REMOTE, GATHER or a COLLECT node where the operation is not COUNT.
+	A subquery also becomes unsuitable if it is contained in an
+	unsuitable subquery.
+
+	The implementation of this feature required reworking the dataflow
+  of query execution.
+
+* Rewrote all modification executors
+
+	as a consequence of the splicing subquery optimisation we rewrote
```
Comment (Simran-B) [inline CHANGELOG]:
```suggestion
	as a consequence of the splicing subquery optimization we rewrote
```

File: CHANGELOG (hunk 4 above)
Comment (goedderz) [reply CHANGELOG]: Agreed. In that case, it's interchangeable, and more understandable.

File: CHANGELOG (hunk 1 above)
Comment (markuspf) [reply CHANGELOG]: This needs to be updated everywhere  (documentation, release notes) as I copy and pasted it from one master version.

**Pull request merged**